pkg/output/gcs/gcs.go

```diff
@@ -0,0 +1,86 @@
+// Licensed to Elasticsearch B.V. under one or more agreements.
+// Elasticsearch B.V. licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+package gcs
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"cloud.google.com/go/storage"
+
+	"github.com/elastic/stream/pkg/output"
+)
+
+func init() {
+	output.Register("gcs", New)
+}
+
+type Output struct {
+	opts       *output.Options
+	client     *storage.Client
+	writer     *storage.Writer
+	cancelFunc func()
+}
+
+func New(opts *output.Options) (output.Output, error) {
+	if opts.Addr == "" {
+		return nil, errors.New("google cloud address is required")
+	}
+	// https://cloud.google.com/go/docs/reference/cloud.google.com/go/storage/latest#hdr-Creating_a_Client
+	// This is required to override the client to use localhost instead, has to be set before creating the client
+	os.Setenv("STORAGE_EMULATOR_HOST", opts.Addr)
+	defer os.Unsetenv("STORAGE_EMULATOR_HOST")
```

Comment:
It looks possible to configure this without setting a global env var via the code in https://github.com/googleapis/google-cloud-go/blob/eddaf71408832097a5514e47a85f19895081dbf9/storage/storage.go#L182-L185. That would be preferable since it doesn't depend on globals.

Reply:
Usually when you see this pattern of calling set and unset, it will restore the state to the previous condition. So if a value was already set then it will be restored and if none was set then it calls unset.